scripts/hud/ground-boost.ts

```diff
@@ -1,56 +1,83 @@
-const TimerFlags = {
-	NONE: 0,
-	LANDING: 1 << 0,
-	KNOCKBACK: 1 << 1,
-	WATERJUMP: 1 << 2
-};
-
-const ColorClass = {
-	FRICTION: 'groundboost__meter--friction',
-	SLICK: 'groundboost__meter--slick',
-	CRASH: 'groundboost__meter--crash'
-};
-
-const LabelClass = {
-	GAIN: 'groundboost__label--gain',
-	FLAT: 'groundboost__label--flat',
-	LOSS: 'groundboost__label--loss',
-	HIDE: 'groundboost__label--hide'
-};
+import { PanelHandler } from 'util/module-helpers';
+import { magnitude2D } from 'util/math';
+import { RegisterHUDPanelForGamemode } from 'util/register-for-gamemodes';
+import { GamemodeCategories, GamemodeCategory } from 'common/web';
+
+enum TimerFlags {
+	NONE = 0,
+	LANDING = 1 << 0,
+	KNOCKBACK = 1 << 1,
+	WATERJUMP = 1 << 2
+}
+
+enum ColorClass {
+	FRICTION = 'groundboost__meter--friction',
+	SLICK = 'groundboost__meter--slick',
+	CRASH = 'groundboost__meter--crash'
+}
+
+enum LabelClass {
+	GAIN = 'groundboost__label--gain',
+	FLAT = 'groundboost__label--flat',
+	LOSS = 'groundboost__label--loss',
+	HIDE = 'groundboost__label--hide'
+}
 
 const MAX_TIMER_MS = 250;
 const MS_2_DEG = 360 / MAX_TIMER_MS;
 const IDEAL_END_MS = 40; // 5 ticks
 
-class Groundboost {
-	/** @type {Panel} @static */
-	static groundboostMeter = $('#GroundboostMeter');
-	/** @type {Label} @static */
-	static groundboostLabel = $('#GroundboostLabel');
-	/** @type {Panel} @static */
-	static container = $('#GroundboostContainer');
+@PanelHandler()
+class GroundboostHandler {
+	readonly panels = {
+		container: $<Panel>('#GroundboostContainer'),
+		groundboostMeter: $<Panel>('#GroundboostMeter'),
+		groundboostLabel: $<Label>('#GroundboostLabel')
+	};
+
+	visible: boolean;
+	overshootEnable: boolean;
+	crashHlEnable: boolean;
+	colorClass: ColorClass;
+	labelClass: LabelClass;
+	missedJumpTimer = 0;
+	peakSpeed = 0;
+	startSpeed = 0;
+	textMode: number;
+	textColorMode: number;
+
+	constructor() {
+		RegisterHUDPanelForGamemode({
+			gamemodes: GamemodeCategories.get(GamemodeCategory.DEFRAG),
+			onLoad: () => this.onMapInit(),
+			events: [
+				{ event: 'HudProcessInput', callback: () => this.onHudUpdate() },
+				{ event: 'OnDefragHUDGroundboostChange', callback: () => this.onConfigChange() }
```

Comment:
for uniformity, should these config events be included in gamemode registration everywhere?

Reply:
Oh, I think this is wrong and should be outside of the gamemode-specific stuff, so updates still happen if not in the given modes. Unless configs are always fetched on map load, will look at that when on my PC